### PR TITLE
[WIP] Setting up API End points for the Post Model

### PR DESCRIPTION
**Pull request by Saurav-Shrivastav**
Still need to add authentication to the end points

### CODE REVIEW

File: buying_selling/urls.py [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
-from django.urls import path
+from django.urls import path, include
 from django.conf import settings
 from django.conf.urls.static import static
 from users import views as user_views
 from rest_framework_simplejwt.views import TokenRefreshView
 
 urlpatterns = [
-    path("admin/", admin.site.urls),
+    path('api-auth/', include('rest_framework.urls')),
```
Comment (jsparmani) [inline buying_selling/urls.py:9]: This line is probably not required. Verify that once

File: posts/api/serializers.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            'brand',
+            'condition',
+        ]
+
+
+class PostDetailSerializer(ModelSerializer):
+    class Meta:
+        model = Post
+        fields = [
+            'id',
```
Comment (jsparmani) [inline posts/api/serializers.py:28]: Would this `id` field be the _UUID_ field?

File: posts/api/urls.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from django.urls import path
+
+from .views import PostCreateAPIView, PostDestroyAPIView, PostDetailAPIView, PostListAPIView, PostUpdateAPIView
+
+urlpatterns = [
+    path('', PostListAPIView.as_view(), name='list'),
+    path('create/', PostCreateAPIView.as_view(), name='create'),
+    path('<pk>/', PostDetailAPIView.as_view(), name='detail'),
```
Comment (jsparmani) [inline posts/api/urls.py:8]: Using `pk` won't really be a good option, we need to use `uuidv4` fields to manage posts.

File: posts/api/serializers.py (hunk 2 above)
Comment (Saurav-Shrivastav) [reply posts/api/serializers.py:28]: Yes exactly

File: posts/api/urls.py (hunk 3 above)
Comment (Saurav-Shrivastav) [reply posts/api/urls.py:8]: The UUID is itself the Pk for the Post model

File: buying_selling/urls.py (hunk 1 above)
Comment (Saurav-Shrivastav) [reply buying_selling/urls.py:9]: Yes it isn't required, I'll remove it

File: posts/api/views.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    return jwt.decode(encoded_token, secret_base_encoded=False, algorithms=['HS256'])
+
+
+class PostCreateAPIView(CreateAPIView):
+    queryset = Post.objects.all()
+    serializer_class = PostCreateSerializer
+    permission_classes = (IsAuthenticated,)
+
+    def perform_create(self, serializer):
+        payload = jwt_decoder(self.request.headers['Token'])
```
Comment (jsparmani) [inline posts/api/views.py:19]: Getting an error on this line. I guess `self.request.headers['Token']` is incorrect. It should be taken from corresponding `Authorization` header.

Comment (jsparmani) [reply posts/api/views.py:19]: The `Authorization` header might require `split` as it has _Bearer_ as prefix to it

**Comment by jsparmani** [comment]
Also, I guess one more route which is required is for the user to fetch the posts that are created by him.

File: posts/api/views.py (hunk 4 above)
Comment (Saurav-Shrivastav) [reply posts/api/views.py:19]: You will have to pass two keys in the header, 
1. Authorization: Normal access token
2. token: The access token without a base 64 encoded secret

**Comment by Saurav-Shrivastav** [comment]
> Also, I guess one more route which is required is for the user to fetch the posts that are created by him.

Yes on it

File: posts/api/views.py (hunk 4 above)
Comment (jsparmani) [reply posts/api/views.py:19]: Okay, so how that secret is getting verified with the backend actually?

Comment (jsparmani) [reply posts/api/views.py:19]: Like, there has to some `secret` fixed on the backend too, how are you planning to do that?

Comment (Saurav-Shrivastav) [reply posts/api/views.py]: I have fixed this in the newer commits!

**Pull request merged**